test/com/esotericsoftware/kryo/serializers/GenericsTest.java

```diff
@@ -185,15 +184,19 @@ public void testClassHierarchyWithMultipleTypeVariables () {
 
 	// Test for https://github.com/EsotericSoftware/kryo/issues/721
 	@Test
-	@Ignore("Currently fails")
 	public void testClassHierarchyWithConflictingTypeVariables () {
 		ClassWithConflictingTypeArguments.A o = new ClassWithConflictingTypeArguments.A(
-				new ClassWithConflictingTypeArguments.B<>(1));
-
-		kryo.setRegistrationRequired(false);
+			new ClassWithConflictingTypeArguments.B<>(1));
 
-		Output buffer = new Output(512, 4048);
-		kryo.writeClassAndObject(buffer, o);
+		try {
+			kryo.setOptimizedGenerics(false);
+			kryo.setRegistrationRequired(false);
+			Output buffer = new Output(512, 4048);
+			kryo.writeClassAndObject(buffer, o);
+		} finally {
+			// reset default
+			kryo.setOptimizedGenerics(true);
+		}
```

Comment:
Could you please replace this with `roundTrip(length, o, false)`? I don't know why I didn't use `roundtrip` when I added the test.

Reply:
I just realized that I didn't use roundtrip because the test class does not implement `equals/hashcode`. Please ignore. I'll improve this test later.